Framework/Indexing/src/IndexInfo.cpp

```diff
@@ -309,6 +306,11 @@ IndexInfo::globalSpectrumIndicesFromDetectorIndices(
       auto bytes = static_cast<int>(sizeof(int64_t) * spectrumIndices.size());
       communicator().send(rank, tag, buffer, bytes);
     }
+    if (std::any_of(detectorMap.begin(), detectorMap.end(),
+                    [](char c) { return c == 1; })) {
+      throw std::runtime_error("Some of the requested detectors do not have a "
+                               "corresponding spectrum");
```

Comment:
How does this work? `c == 1` should indicate that there is a mapping? Should it read `c == 0`? But that would not work either: `detectorMap` lists *all* detectors, but you want to have such a check only for the requested values in `detectorIndices`.

Reply:
If there is a mapping, `detectorMap[detectorIndex]` should be 2 at this point, since it would have been incremented above. So c==1 here means that there was a detector in `detectorIndices` that had no mapping. If there is a single detector like that, we throw, following our slack discussion (see next comment).

Reply:
:+1: